content/docs/hooks-intro.md

```diff
@@ -25,86 +25,86 @@ function Example() {
 
-## Motivation {#motivation}
+## ការជំរុញលើកទឹកចិត្ត {#motivation}
 
-Hooks solve a wide variety of seemingly unconnected problems in React that we've encountered over five years of writing and maintaining tens of thousands of components. Whether you're learning React, use it daily, or even prefer a different library with a similar component model, you might recognize some of these problems.
+Hooks ដោះស្រាយ a wide variety of seemingly unconnected problems នៅក្នុង React ដែលពួកយើងបានជួបប្រទះជាងប្រាំឆ្នាំនៃការសរសេរ ក៏ដូចជាការ maintaining រាប់ពាន់ components។ ដូច្នេះមិនថាអ្នកកំពុងរៀន React, កំពុងប្រើប្រាស់វា ឬក៏ប្រើ library ផ្សេងដែលស្រដៀងនិង component model របស់ React ក៏ដោយ, អ្នកប្រហែលជាធ្លាប់ស្គាល់បញ្ហាទាំងនេះ។
 
-### It's hard to reuse stateful logic between components {#its-hard-to-reuse-stateful-logic-between-components}
+### វាពិតជាពិបាកខ្លាំងមែនទែននៅពេលដែលចង់ប្រើ stateful logic ជាមួយនឹង components ផ្សេងទៀត {#its-hard-to-reuse-stateful-logic-between-components}
 
-React doesn't offer a way to "attach" reusable behavior to a component (for example, connecting it to a store). If you've worked with React for a while, you may be familiar with patterns like [render props](/docs/render-props.html) and [higher-order components](/docs/higher-order-components.html) that try to solve this. But these patterns require you to restructure your components when you use them, which can be cumbersome and make code harder to follow. If you look at a typical React application in React DevTools, you will likely find a "wrapper hell" of components surrounded by layers of providers, consumers, higher-order components, render props, and other abstractions. While we could [filter them out in DevTools](https://github.com/facebook/react-devtools/pull/503), this points to a deeper underlying problem: React needs a better primitive for sharing stateful logic.
+React មិនបានផ្តល់នូវវិធី "attach" reusable behavior ទៅនិង component ទេ (ឧទាហរណ៍, connecting it to a store). ប្រសិនបើអ្នកធ្វើការជាមួយ React ពីមុនមក, អ្នកប្រហែលជាធ្លាប់ឃើញ patterns ខ្លះៗដូចជា [render props](/docs/render-props.html) ហើយនិង [higher-order components](/docs/higher-order-components.html) ជាដើម ដែលវាបានព្យាយាមដោះស្រាយបញ្ហាទាំងអស់នេះ។ ក៏ប៉ុន្តែ patterns ទាំងនេះតម្រូវអោយ restructure components របស់អ្នកនៅពេលដែលត្រូវការវា, ដែលវាធ្វើអោយកូដមើលទៅរៀងរញ៉េរញ៉ៃហើយពិបាកនិងតាម. ប្រសិនបើអ្នកក្រឡេកទៅមើល a typical React application នៅក្នុង React DevTools, អ្នកប្រាកដជាឃើញ a "wrapper hell" នៃ components ដែលហុំព័ទ្ធទៅដោយ layers of providers, consumers, higher-order components, render props, ហើយនិង abstractions មួយចំនួន។ អំឡុងពេលដែលយើងអាច  [filter them out in DevTools](https://github.com/facebook/react-devtools/pull/503), ត្រង់នេះហើយដែលជាចំនុចរសើបនៃ បញ្ហាដែលកប់នៅក្នុង React។ ដូច្នេះហើយ React ត្រូវការកែកុនអោយប្រសើរជាមុនក្នុងការ sharing stateful logic។
```

Comment:
(ឧទាហរណ៍, connecting it to a store).  => (ឧទាហរណ៍, connecting it to a store)។
រញ៉េរញ៉ៃហើយពិបាកនិងតាម. => រញ៉េរញ៉ៃហើយពិបាកនិងតាម។

Reply:
I think we should not use `,` in khmer translation